src/extended.rs

```diff
@@ -1770,3 +1773,694 @@ bitflags! {
+    pub fn has_cpuid_user_dis(&self) -> bool {
+        self.eax
+            .contains(ExtendedFeatureIdentification2Eax::CPUID_USER_DIS)
+    }
+
+    /// The size of the Microcode patch in 16-byte multiples. If 0, the size of the
+    /// patch is at most 5568 (15C0h) bytes.
+    pub fn microcode_patch_size(&self) -> u16 {
+        get_bits(self.ebx, 0, 11) as u16
+    }
+}
+
+bitflags! {
+    #[repr(transparent)]
+    #[derive(Debug, Clone, Copy, PartialEq, Eq)]
+    struct ExtendedFeatureIdentification2Eax: u32 {
+        const NO_NESTED_DATA_BP = 1 << 0;
+        const LFENCE_ALWAYS_SERIALIZING = 1 << 2;
+        const SMM_PG_CFG_LOCK = 1 << 3;
+        const NULL_SELECT_CLEARS_BASE = 1 << 6;
+        const UPPER_ADDRESS_IGNORE = 1 << 7;
+        const AUTOMATIC_IBRS = 1 << 8;
+        const NO_SMM_CTL_MSR = 1 << 9;
+        const PREFETCH_CTL_MSR = 1 << 13;
+        const CPUID_USER_DIS = 1 << 17;
+    }
+}
+
+/// Extended Feature Identification 2 (LEAF=0x8000_0021).
+///
+/// # Platforms
+/// ✅ AMD ❌ Intel
+#[derive(PartialEq, Eq, Debug)]
+pub struct ExtendedPerformanceMonitoringDebug {
+    eax: ExtendedPerformanceMonitoringDebugEax,
+    ebx: u32,
+    _ecx: u32,
+    _edx: u32,
+}
+
+impl ExtendedPerformanceMonitoringDebug {
+    pub(crate) fn new(data: CpuIdResult) -> Self {
+        Self {
+            eax: ExtendedPerformanceMonitoringDebugEax::from_bits_truncate(data.eax),
+            ebx: data.ebx,
+            _ecx: data.ecx,
+            _edx: data.edx,
+        }
+    }
+
+    /// Performance Monitoring Version 2 supported if set
+    pub fn has_perf_mon_v2(&self) -> bool {
+        self.eax
+            .contains(ExtendedPerformanceMonitoringDebugEax::PERF_MON_V2)
+    }
+
+    /// Last Branch Record Stack supported if set
+    pub fn has_lbr_stack(&self) -> bool {
+        self.eax
+            .contains(ExtendedPerformanceMonitoringDebugEax::LBR_STACK)
+    }
+
+    /// Freezing Core Performance Counters and LBR Stack on Core
+    /// Performance Counter overflow supported if set
+    pub fn has_lbr_and_pmc_freeze(&self) -> bool {
+        self.eax
+            .contains(ExtendedPerformanceMonitoringDebugEax::LBR_AND_PMC_FREEZE)
+    }
+
+    /// Number of Core Performance Counters
+    pub fn num_perf_ctr_core(&self) -> u8 {
+        get_bits(self.ebx, 0, 3) as u8
+    }
+
+    /// Number of Last Branch Record Stack entries
+    pub fn num_lbr_stack_size(&self) -> u8 {
+        get_bits(self.ebx, 4, 9) as u8
+    }
+
+    /// Number of Northbridge Performance Monitor Counters
+    pub fn num_perf_ctr_nb(&self) -> u8 {
+        get_bits(self.ebx, 10, 15) as u8
+    }
+}
+
+bitflags! {
+    #[repr(transparent)]
+    #[derive(Debug, Clone, Copy, PartialEq, Eq)]
+    struct ExtendedPerformanceMonitoringDebugEax: u32 {
+        const PERF_MON_V2 = 1 << 0;
+        const LBR_STACK = 1 << 1;
+        const LBR_AND_PMC_FREEZE = 1 << 2;
+    }
+}
+
+/// Extended Feature Identification 2 (LEAF=0x8000_0021).
```

Comment:
this and a few doc comments on other new structs describing leaves `8000_002[236]` above are copy-pasted and incorrect. the `fn get_*` below that read these leaves look correct, so probably just reuse that text on the structs too?